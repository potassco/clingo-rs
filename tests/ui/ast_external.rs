// ast_external
use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let atom1 = ast::symbolic_atom(term1).unwrap();
    let body = [];
    let ext = ast::external(&loc, atom1, &body, ExternalType::Free);
    drop(body);
    let _end = ext;
}
