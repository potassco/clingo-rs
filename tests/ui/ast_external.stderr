error[E0505]: cannot move out of `body` because it is borrowed
  --> tests/ui/ast_external.rs:11:10
   |
9  |     let body = [];
   |         ---- binding `body` declared here
10 |     let ext = ast::external(&loc, atom1, &body, ExternalType::Free);
   |                                          ----- borrow of `body` occurs here
11 |     drop(body);
   |          ^^^^ move out of `body` occurs here
12 |     let _end = ext;
   |                --- borrow later used here
